operator/v1alpha1/types_image_content_source_policy.go

```diff
@@ -0,0 +1,45 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImageContentSourcePolicy holds cluster-wide information about how to handle registry mirror rules.  The canonical name is `cluster`
+type ImageContentSourcePolicy struct {
```

Comment:
Any other possible names we could use?  Policy is fairly common.  Is `content source` the best name?  I don't want to be specific about "mirroring" or "disconnected", so trying out a name that implies "you can change where we source images" was the initial argument.

Maybe future might be "you have to have signatures from X for Y"

Reply:
Just `ImageContentSource` or `ImageSources`?

Reply:
Changed it to `ImageContentSource`

Reply:
By "policy is fairly common" I meant "It's good to have Policy in there to distinguish this from a more concrete object".  I don't think you have to change the name, just want to see some suggestions about what names could be that better describe what this object does.

Reply:
changed the name back

Reply:
You’re right that “content source” is not a commonly used term for this. I’d just say one of “image {repositories,locations,(servers,storage?)}”; the “content” word adds nothing.

---

Per https://github.com/openshift/api/pull/354#discussion_r295558500:
> Do we want to document this as a general-purpose facility, or specifically restricted to OpenShift product images? 

do we want this to be generic for all kinds of images, or actually say something entirely specific to the contemplated use, in the ballpark of `OpenShiftReleasedImageLocations`?

Reply:
I think making it generic will allow us to reuse it for other use cases in the future.

Reply:
The resource group usually contains the product namespace (operator.openshift.io) so it is redundant in the title.